test/token/ERC721/extensions/ERC721Consecutive.test.js

```diff
@@ -1,222 +1,244 @@
           batches.map(({ receiver }) => receiver),
           batches.map(({ amount }) => amount),
-        );
+        ]);
+
+        return { accounts, alice, bruce, chris, receiver, batches, delegates, token };
+      }
+
+      beforeEach(async function () {
+        Object.assign(this, await loadFixture(fixture));
       });
 
       describe('minting during construction', function () {
         it('events are emitted at construction', async function () {
           let first = offset;
-
-          for (const batch of batches) {
+          for (const batch of this.batches) {
             if (batch.amount > 0) {
-              await expectEvent.inConstruction(this.token, 'ConsecutiveTransfer', {
-                fromTokenId: web3.utils.toBN(first),
-                toTokenId: web3.utils.toBN(first + batch.amount - 1),
-                fromAddress: constants.ZERO_ADDRESS,
-                toAddress: batch.receiver,
-              });
+              await expect(this.token.deploymentTransaction())
+                .to.emit(this.token, 'ConsecutiveTransfer')
+                .withArgs(
+                  first /* fromTokenId */,
+                  first + batch.amount - 1n /* toTokenId */,
+                  ethers.ZeroAddress /* fromAddress */,
+                  batch.receiver.address /* toAddress */,
+                );
             } else {
-              // expectEvent.notEmitted.inConstruction only looks at event name, and doesn't check the parameters
+              // ".to.not.emit" doesn't check the parameters
+              // await expect(this.token.deploymentTransaction())
+              //   .to.not.emit(this.token, 'ConsecutiveTransfer')
+              //   .withArgs(
+              //     first, /* fromTokenId */
+              //     first + batch.amount - 1n, /* toTokenId */
+              //     ethers.ZeroAddress, /* fromAddress */
+              //     batch.receiver, /* toAddress */
+              //   );
             }
             first += batch.amount;
           }
         });
 
         it('ownership is set', async function () {
           const owners = [
-            ...Array(offset).fill(constants.ZERO_ADDRESS),
-            ...batches.flatMap(({ receiver, amount }) => Array(amount).fill(receiver)),
+            ...Array(Number(offset)).fill(ethers.ZeroAddress),
+            ...this.batches.flatMap(({ receiver, amount }) => Array(Number(amount)).fill(receiver.address)),
           ];
 
           for (const tokenId in owners) {
-            if (owners[tokenId] != constants.ZERO_ADDRESS) {
-              expect(await this.token.ownerOf(tokenId)).to.be.equal(owners[tokenId]);
+            if (owners[tokenId] != ethers.ZeroAddress) {
+              expect(await this.token.ownerOf(tokenId)).to.equal(owners[tokenId]);
             }
           }
         });
 
         it('balance & voting power are set', async function () {
-          for (const account of accounts) {
+          for (const account of this.accounts) {
```

Comment:
This test relies on `this.accounts` being equal to `await ethers.getSigners()`. I think there's a chance we (or any contributor) forget and only pass a subset of the accounts to this test.

I don't have a clear suggestion, but wanted to point out. Maybe we add a comment in the fixture?

Reply:
this loop checks many accounts. This includes account with token, accounts without token, accounts with delegation and accounts without delegation.

If the scope of the loop was reduced, it would still pass, but coverage would possibly go down.

I see two alternatives:
- loop over an explicit set of signers 
```
for (const accounts of [this.alice, this.bruce, this.chris, this.receiver]) {
```
- loop over all signers explicitelly
```
for (const accounts of await ethers.getSigners()) {
```

WDYT?

Reply:
Let's do the second one, I checked and the speed difference in the tests is almost none.